### ISSUE TITLE
Working directory and paths in run log tibble are correct

### ISSUE THREAD

**Issue by seth127** [metrumresearchgroup/rbabylon#71]
# Summary
The `run_log()` functions has some relative paths and some absolute paths. This is causing an issue for portability (i.e. looking at results when the absolute path is different than it was when the model was run). 

This may just require some guidance on how to use certain fields in the run log, but we should look into it.

As reported by @callistosp --

It looks like absolute paths are used for `orig_working_dir` in the run log and `output_dir` in the config log. This is causing some portability issues that I noticed when working with Adam on some stuff today. Is there a reason these aren’t relative paths? 

Here’s an example. This is just a head from the example project
```
   run_id orig_working_dir                             output_dir                                                     
   <chr>  <chr>                                        <chr>                                                          
 1 100    /data/rbabylon-example-project/model/pk      /data/home/adaml/Projects/rbabylon-example-project/model/pk/100
```

The first issue is that orig_working_dir and output_dir don’t match.

The second issue is that if I want to use `output_dir` to generate, for example, the hash of the CTL to compare with the hash from the model when it was run, I can’t use `output_dir` to easily point to the file across instances

# Tests
- tests/testthat/test-run-log.R
  - run_log() works correctly with nested dirs
  - config_log() works correctly with nested dirs
  - add_config() works correctly with nested dirs
- tests/testthat/test-workflow-bbi.R
  - config_log() works correctly
  - run_log() captures runs correctly
  - add_config() md5 matches original md5

**Comment by dpastoor** [comment]
so this is an interesting one - I brought this up with @shairozan. There are a number of things going on here, some expected/desired, others maybe buggy. The reason we chose absolute paths is it really codifies exactly where this was run - in particular it was run in this case on _adams_ machine. 

There are some tricks that can be used to then normalize that, especially in the fs package - but I also see it looks like the original_working_dir vs the output dir don't seem to make sense - how did it start in /data then output into his project dir in his home directory.


This can also play out a bit unexpectedly when doing SGE submission. In particular, the way SGE submission works is its a 2 pass mechanism - pass 1 submits a job to run "locally" on the worker node. 


I'd like to find out more specifically of exactly what he's running.

**Comment by callistosp** [comment]
Just to clarify this is in an example project that Adam and I have been collaboratively working on (mostly Adam). On _his machine_, the `orig_working_dir` and `output_dir` roots match. However, this output is from _my machine_, where the `orig_working_dir` matches the root of how my project is setup.

The problem we are having is that we want both of us to be able to run the same script to generate a run log from our own machines without the need to hardcode file paths.

**Comment by seth127** [comment]
@callistosp and Adam (what is Adam's github name?) I'm gonna start looking into this soon. Can you give me a little more detail on what you were trying to accomplish? 

I understand the issue is because the model run and post-run activities were happening on different machines, but what _are_ those post-run activities? Were you basically trying to do what's described here? https://github.com/metrumresearchgroup/rbabylon/issues/30

Was there anything else you were trying to accomplish that was having this same issue? Thanks.

**Comment by callistosp** [comment]
Yeah, @ adamlafontaine and I were trying to implement what is described in #30. There may be other areas this would cause an issue, but for now just looking at differences in hash for models/data sets.

**Comment by seth127** [comment]
Ok, some interesting findings here. First of all, you can get to what you want by doing this:

```
rlog <- run_log() %>% left_join(config_log(), by = "run_id")
norm_paths <- fs::path_norm(file.path(rlog$orig_working_dir, rlog$run_id, rlog$data_path))

rlog <- rlog %>% mutate(current_data_md5 = tools::md5sum(norm_paths),
                current_model_md5 = tools::md5sum(file.path(getOption("rbabylon.model_directory"), model_path))) 

# this just shows the relevant columns, but you could also do the filtering thing Devin had originally
rlog %>% select(orig_working_dir, run_id, data_md5, current_data_md5, model_md5, current_model_md5)
```

The key thing here is that we're _not_ using the `output_dir` column that comes from the `config_log()` output and instead using `file.path(rlog$orig_working_dir, rlog$run_id)` to get the output directory. 

This is because the `config_log()` is intentionally persisting information about the state at the time of the actual model being run. By contrast `orig_working_dir` (a name I've grown to hate, btw) is persisting the absolute path to the model _when the YAML is read into memory_ each time--in this case, when you call `run_log()`.

That's probably enough to digest in one comment. I have further thoughts that I'll put in another comment...

**Comment by seth127** [comment]
After looking at this, I have two important changes that I think should be made. I already have the ad-hoc code, so I'll prepare a branch for review and we can decide if we want to merge it or not.

# Thing 1: We should include less fields in config_log
Specifically we should drop `output_dir` (because of the absolute path confusion) and `model_name` (because it's pointless and can only cause accidental confusion). These can still be parsed out of the actual `bbi_config.json` if someone wants to do some detective work about the state of the machine when the model was actually run, but I think they're more-likely-than-not only going to cause trouble for post-processing.

# Thing 2: We need to revamp both log functions treatment of paths
This came up with @dpastoor and I recently. Specifically, when there are nested hierarchical modeling directories, some things break down. The `run_log()` and `config_log()` are two of those things. 

We do a lot based on joining by `run_id` but when you have `model/pk/101` and also `model/placebo/101` and also `model/pk/bootstrap/101` then they all have `run_id` of `101` and things get ugly.

I have some ideas to fix this in the works (and the incoming PR). Con: they would be breaking changes for any run_log() code you currently have. Pro: they will fix this issue and also make the md5 checking a lot easier too. Stay tuned.

**Comment by seth127** [comment]
This is fixed as of release 0.5.0 and will close once that merges to master.

**Comment by seth127** [comment]
(original post copied here during validation)

# Summary
The `run_log()` functions has some relative paths and some absolute paths. This is causing an issue for portability (i.e. looking at results when the absolute path is different than it was when the model was run). 

This may just require some guidance on how to use certain fields in the run log, but we should look into it.

As reported by @callistosp --

It looks like absolute paths are used for `orig_working_dir` in the run log and `output_dir` in the config log. This is causing some portability issues that I noticed when working with Adam on some stuff today. Is there a reason these aren’t relative paths? 

Here’s an example. This is just a head from the example project
```
   run_id orig_working_dir                             output_dir                                                     
   <chr>  <chr>                                        <chr>                                                          
 1 100    /data/rbabylon-example-project/model/pk      /data/home/adaml/Projects/rbabylon-example-project/model/pk/100
```

The first issue is that orig_working_dir and output_dir don’t match.

The second issue is that if I want to use `output_dir` to generate, for example, the hash of the CTL to compare with the hash from the model when it was run, I can’t use `output_dir` to easily point to the file across instances
